packages/styled-components/src/models/StyledNativeComponent.js

```diff
@@ -79,7 +79,7 @@ class StyledNativeComponent extends Component<*, *> {
             const themeProp = determineTheme(this.props, theme, defaultProps);
```

Comment:
I'm wondering if this line should be pulled up above L78 and then use `themeProp` in L82 where you changed it to `this.props.theme`.

Reply:
I tried that first, but then I also had to change determineTheme, to support undefined as a theme. And for completeness you also need to use it like that in the respective place in `StyledComponent.js`. It does allow you to remove the if else statement in both places. I wanted to keep this PR small, to make it easy to get it reviewed and this bug fixed. So I went for the simple change that is the same as in `StyledComponent.js`

Reply:
Well we never want the theme value to be undefined, it should always be at least an empty object.